server/src/test/java/org/elasticsearch/cluster/routing/allocation/ExpectedShardSizeAllocationTests.java

```diff
@@ -92,43 +76,26 @@ public Long getShardSize(ShardRouting shardRouting) {
 
     public void testExpectedSizeOnMove() {
         final long byteSize = randomIntBetween(0, Integer.MAX_VALUE);
-        final AllocationService allocation = createAllocationService(Settings.EMPTY, () -> new ClusterInfo() {
-            @Override
-            public Long getShardSize(ShardRouting shardRouting) {
```

Comment:
We should ideally stop overriding methods on cluster info instead of creating a valid objects for tests.
This one has failed as allocator was accessing this data via overloaded method and not finding anything.